src/main/java/com/atlassian/db/replica/api/DualConnection.java

```diff
@@ -375,25 +375,45 @@ public PreparedStatement prepareStatement(String sql, String[] columnNames) thro
     @Override
     public Clob createClob() throws SQLException {
         checkClosed();
-        throw new ReadReplicaUnsupportedOperationException();
+        if(compatibleWithPreviousVersion)
+            throw new ReadReplicaUnsupportedOperationException();
```

Comment:
How do you want to roll out and test those changes? Do you have product that will use them ? AFAIK we don't have any way to verify the rollout, by verify I meant use the implementation on production.

Reply:
I guess any implementation will work better than `throw new ReadReplicaUnsupportedOperationException();`. This implementation delegates to a real connection so should be safe. The only question can it use a `read` connection or do we have to use a `write` connection?
I think the main goal is to slowly implement the methods we don't support and remove `ReadReplicaUnsupportedOperationException`.

Reply:
Using `getWriteConnection` will work for sure.
Other methods use `getReadConnection`, which would throw `SQLException` if mis-routed. Tracking `ReadReplicaUnsupportedOperationException` could actually be easier than yet another `SQLException`, but we have to try using replica at some point.

Reply:
We can use `getWriteConnection` as a safe option and think about switching when proofed it hurts efficiency.

Reply:
> I guess any implementation will work better than

Reply:
How do other open-source libraries do it? They simply wait for feedback from consumers and release bugfixes.
Any reason why this strategy won't work for us?

Reply:
Agree.

Reply:
Postgres JDBC driver has a [huge suite of tests](https://github.com/pgjdbc/pgjdbc/blob/v42.3.0-rc5/pgjdbc/src/test/java/org/postgresql/test/jdbc2/Jdbc2TestSuite.java).

Reply:
> How do other open-source libraries do it? They simply wait for feedback from consumers and release bugfixes.

Which lib does this? Every repo I saw asked contributors to include tests.

Reply:
What cases would you like to see being tested?

Reply:
Smoke test by calling these methods in `DualConnectionIT`. It can be a single test case going through all the methods.
The harness is already set up, it should be quite cheap to write.